edx/edx-enterprise
# ENT-455 show consent decline notification on enterprise course enrollment page

@saleem-latif @asadiqbal08 @douglashall 

**Description:** Redirect learner to enterprise course enrollment page after declining consent and display consent decline notification.

**JIRA:** [ENT-455](https://openedx.atlassian.net/browse/ENT-455)

**Dependencies:** N/A

**Merge deadline:** 30 June, 2017

**Installation instructions:** Install `edx-enterprise` from this branch `zub/ENT-455-ent-landing-page-dsc-decline` in `edx-platform`.

**Testing instructions:**

1. Go to enterprise customers django admin at `admin/enterprise/enterprisecustomer/`
2. Add enterprise customer with logo and with consent enabled with option `At Enrollment` and link it with an `idp`.
3. Create/publish a course in studio and create related modes in Ecom course admin tool.
4. As an anonymous user view the enterprise landing page for the enterprise course by accessing the enterprise course enrollment url with enterprise customer uuid from step no 2, e.g: http://zubair.localhost:8000/enterprise/72416e52-8c77-4860-9584-15e5b06220fb/course/course-v1:edX+TEST_01+2016_R1/enroll/
5. The learner will be redirected to `idp` login page related to the enterprise and on successful login learner will be able view enterprise landing page after linking the edX account with enterprise.
6. On clicking the button `"Continue"`, after selecting a course mode the learner will be redirected to a course specific consent page.
7. Clicking on link `"No, take me back."` will show a modal with consent declining message "Are you aware...".
8. Now verify that on clicking the button `"I decline"` the learner is redirected back to enterprise course enrollment page with consent decline warning notification (screenshot attached).
<img width="740" alt="screen shot 2017-06-23 at 2 30 32 pm" src="https://user-images.githubusercontent.com/5072991/27476102-e09f88cc-5820-11e7-90cc-440bbca0e22e.png">



**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
